modules/monitoring-configuring-shiftstack-scraping.adoc

```diff
@@ -0,0 +1,86 @@
+
+You can open the federation endpoint of your {product-title} cluster to make metrics available to a {rhoso-first} cluster to practice pull-based monitoring.
+
+.Prerequisites
+
+- You have administrative access to {rhoso} the tenant cluster that is running on it.
+- Telemetry is enabled in the {rhoso} environment.
+- The Cluster Observability Operator is installed on your cluster.
+- The monitoring stack for your cluster is configured.
+- Your cluster has its federation endpoint exposed.
+// That is, https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/monitoring/accessing-third-party-monitoring-apis#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-monitoring-apis-by-using-the-cli
+
+.Procedure
+
+. Connect to your cluster by using a username and password; do not log in by using a `kubeconfig` file that was generated by the installation program.
+
+. To retrieve a token from the cluster, run the following command:
++
+[source,terminal]
+----
+$ oc whoami -t
+----
+
+. To get the Prometheus federation route URL from your cluster, enter the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring get route prometheus-k8s-federate -ojsonpath={'.status.ingress[].host'}
+----
+
+. Write a manifest for a scrape configuration and save it as a file called `cluster-scrape-config.yaml`. As an example:
++
+[source,yaml]
+----
+apiVersion: monitoring.rhobs/v1alpha1
+kind: ScrapeConfig
+metadata:
+  labels:
+    service: metricStorage
+  name: sos1-federated
+  namespace: openstack
+spec:
+  params:
+    'match[]':
+    - '{__name__=~"kube_node_info|kube_persistentvolume_info"}' # <1>
+  metricsPath: '/federate'
+  authorization:
+    type: Bearer
+    credentials:
+      name: ocp-federated # <2>
+      key: token
+  scheme: HTTPS # or HTTP
+  scrapeInterval: 30s # <3>
+  staticConfigs:
+  - targets:
+    - prometheus-k8s-federate-openshift-monitoring.apps.openshift.example # <4>
+----
+<1> Add metrics here. In this example, only the metrics `kube_node_info` and `kube_persistentvolume_info` are requested.
+<2> Note the name of the secret for later.
+<3> Limit scraping to fewer than 1000 samples for each request with a maximum frequency of once every 30 seconds.
+<4> Insert the URL you fetched previously here. If the endpoint is HTTPS and uses a custom certificate authority, add a `tlsConfig` section after it.
+
+. While connected to the {product-title} cluster that hosts the RHOSO control plane, apply the manifest by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f cluster-scrape-config.yaml
+----
+
+. Make the token available as a secret in the `openstack` namespace by running the following command:
++
+[source,terminal]
+----
+$ oc -n openstack create secret generic ocp-federated --from-literal=token=<the_token_fetched_previously>
+----
++
+Note that `ocp-federated` is the name of the secret in the previous `ScrapeConfig` manifest.
```

Comment:
I would move this step to the third position in the list, just after the `$ oc whoami -t` step. Otherwise we would be creating the `scrapeconfig` object pointing to a secret (`ocp-federated`) that doesn't exist, which would probably fail.
As the creation of the secret needs to be done in the RHOSO's management OCP cluster I would clearly state that as there are other steps that need to be done in the shift-on-stack cluster.